src/model/openapi31.ts

```diff
@@ -265,6 +265,7 @@ export type SchemaObjectType =
     | 'array';
 
 export interface SchemaObject extends ISpecificationExtension {
+    $ref?: string,
```

Comment:
Might technically break your `isReferenceObject` but SchemaObjects can also contain $ref so 🤷

Reply:
Indeed! You are right.